cmd/instance_update.go

```diff
@@ -25,6 +26,7 @@ type instanceUpdateCmd struct {
 	CloudInitCompress bool              `cli-flag:"cloud-init-compress" cli-usage:"compress instance cloud-init user data"`
 	Labels            map[string]string `cli-flag:"label" cli-usage:"instance label (format: key=value)"`
 	Name              string            `cli-short:"n" cli-usage:"instance name"`
+	Protection        bool              `cli-flag:"protection" cli-usage:"enable delete protection"`
```

Comment:
Unfortunately one flag is not enough here, you'll need to create two flags `--add-protection` and `--remove-protection`.
The problem with your current implementation, is that it always removes the instance protection whenever you run an update command without the `--protection` flag. That's not ideal, users should have to set `--remove-protection` explicitly to do that. When the user doesn't set a protection flag, nothing related to delete protection should happen.

Reply:
There is a bug there for sure, but we don't need a second flag.
We can use `--protection=false` for removing delete protection (same how we manage DBaaS `--termination-protection`).
There is function to check if flag was specified at all (see [here](https://github.com/exoscale/cli/blob/master/cmd/dbaas_update_pg.go#L60C2-L60C79)).